tests/41end-to-end-keys/07-backup.pl

```diff
@@ -0,0 +1,387 @@
+         method  => "PUT",
+         uri     => "/unstable/room_keys/keys/!abcd/1234",
+         params  => {
+            version => $current_version,
+         },
+         content => {
+            first_message_index => 3,
+            forwarded_count     => 0,
+            is_verified         => JSON::false,
+            session_data        => "anopaquestring",
+         }
+      )->then( sub {
+         my ( $content ) = @_;
+         log_if_fail "Content", $content;
+
+         do_request_json_for( $user,
+            method  => "GET",
+            uri     => "/unstable/room_keys/keys/!abcd/1234",
+            params  => {
+               version => $current_version,
+            }
+         );
+      })->then( sub {
+         my ( $content ) = @_;
+         log_if_fail "Content", $content;
+
+         assert_json_keys( $content, "first_message_index" );
+
+         assert_json_keys( $content, "forwarded_count" );
+
+         assert_json_keys( $content, "is_verified" );
+
+         assert_json_keys( $content, "session_data" );
+
+         $content->{first_message_index} == 3 or
+            die "Expected first message index to match submitted data";
+
+         $content->{forwarded_count} == 0 or
+            die "Expected forwarded count to match submitted data";
+
+         $content->{is_verified} == JSON::false or
+            die "Expected is_verified to match submitted data";
+
+         $content->{session_data} eq "anopaquestring" or
+            die "Expected session data to match submitted data";
+
+         Future->done(1);
+      });
+   };
+
+test "Can update keys with better versions",
+   requires => [ $fixture, qw( can_backup_e2e_keys ) ],
+
+   proves => [qw( can_update_e2e_keys )],
+
+   do => sub {
+      my ( $user ) = @_;
+
+      do_request_json_for( $user,
+         method  => "PUT",
+         uri     => "/unstable/room_keys/keys/!abcd/1234",
+         params  => {
+            version => $current_version,
+         },
+         content => {
+            first_message_index => 1,
+            forwarded_count     => 0,
+            is_verified         => JSON::false,
+            session_data        => "anotheropaquestring",
+         }
+      )->then( sub {
+         my ( $content ) = @_;
+         log_if_fail "Content", $content;
+
+         do_request_json_for( $user,
+            method  => "GET",
+            uri     => "/unstable/room_keys/keys/!abcd/1234",
+            params  => {
+               version => $current_version,
+            }
+         );
+      })->then( sub {
+         my ( $content ) = @_;
+         log_if_fail "Content", $content;
+
+         assert_json_keys( $content, "first_message_index" );
+
+         assert_json_keys( $content, "forwarded_count" );
+
+         assert_json_keys( $content, "is_verified" );
+
+         assert_json_keys( $content, "session_data" );
+
+         $content->{first_message_index} == 1 or
+            die "Expected first message index to match submitted data";
```

Comment:
I think you want to use `assert_eq( $content->{first_message_index}, 1, ".." )` as it will have better logging on failures (in terms of logging what the actual values were?)